app/src/main/java/com/rage/clamber/Activities/HomePage.java

```diff
@@ -34,49 +43,131 @@ protected void onCreate(Bundle savedInstanceState) {
+            transaction.replace(R.id.home_page_frame_layout, WallsFragment.newInstance(user));
+            transaction.addToBackStack(null);
+            transaction.commit();
+        }
+        else if (tab.getPosition() == 2){
+            clearBackstack();
+            FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
+            transaction.replace(R.id.home_page_frame_layout, ProjectsFragment.newInstance(user));
+            transaction.addToBackStack(null);
+            transaction.commit();
+        }
+        else if (tab.getPosition() == 3) {
+            clearBackstack();
+            FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
+            transaction.replace(R.id.home_page_frame_layout, UserInfoFragment.newInstance(user));
+            transaction.addToBackStack(null);
+            transaction.commit();
+        }
     }
 
-    @OnClick(R.id.action_bar_walls_button)
-    public void onWallsButtonClicked(Button button) {
-        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
-        transaction.replace(R.id.home_page_frame_layout, WallsFragment.newInstance(user));
-        transaction.addToBackStack(null);
-        transaction.commit();
+    /**
+     * On back pressed method - created to exit application instead of hitting the login page. Checks
+     * to see if there are no items in the backstack (indicitive of being back at the Home Activity).
+     * If that is the case, it exits the application. It also sets the tab selected to the Home tab
+     * if the backstack is at one.
+     */
+    @Override
+    public void onBackPressed() {
+
+        if (getSupportFragmentManager().getBackStackEntryCount() == 0) {
+            Intent intent = new Intent(Intent.ACTION_MAIN);
+            intent.addCategory(Intent.CATEGORY_HOME);
+            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            startActivity(intent);
+            finish();
+        }
+        else if (getSupportFragmentManager().getBackStackEntryCount()==1){
+            TabLayout.Tab tab = tabLayout.getTabAt(0);
+            tab.select();
+            getSupportFragmentManager().popBackStack();
+        }
+        else {
+            getSupportFragmentManager().popBackStack();
+        }
     }
 
-    @OnClick(R.id.action_bar_projects_button)
-    public void onProjectsButtonClicked(Button button) {
-        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
-        transaction.replace(R.id.home_page_frame_layout, ProjectsFragment.newInstance(user));
-        transaction.addToBackStack(null);
-        transaction.commit();
+    /**
+     * Method to clear the backstack. Used to ensure fragments are not building up. Called each time
+     * a tab is changed.
+     */
+    private void clearBackstack() {
+        FragmentManager manager = getSupportFragmentManager();
+        if (manager.getBackStackEntryCount() > 0) {
```

Comment:
It seems like this just removes the first element from the backstack, is that intention, or does `manager.popBackStack` do something other than just remove the specific backstack entry you give it?